addons/first_heroes_win.wad/init.lua

```diff
@@ -0,0 +1,184 @@
+
+local debug = false
+local function d_print(text)
+   if debug then print(text) end
+end
+local function t_print(tbl)
+   if debug then
+      local t = ""
+      for i,v in ipairs(tbl) do
+         t = t .. v .. ", "
+      end
+      return "{"..t.."}"
+   end
+   return ""
+end
+
+-- If changing this value remember to change it also in the addons-file
+local MAXHEROES = 4
+
+push_textdomain("first_heroes_win.wad", true)
+
+local wc_name = "First Heroes Win"
+-- This needs to be exactly like wc_name, but localized, because wc_name
+-- will be used as the key to fetch the translation in C++
+local wc_descname = _("First Heroes Win")
+local wc_version = 1
+local wc_desc = _("The player who has at least %1% fully trained soldiers in his warehouses wins the game."):bformat(MAXHEROES)
+local wc_first_hero = "Number of heroes"
+-- This needs to be exactly like wc_first_hero, but localized, because wc_first_hero
+-- will be used as the key to fetch the translation in C++
+local wc_first_hero_i18n = _("Number of heroes")
+
+local r = {
+   name = wc_name,
+   description = wc_desc,
+   peaceful_mode_allowed = true,
+   func = function()
+      local plr_data = {}
+      local plrs = wl.Game().players
+      for i, plr in ipairs(plrs) do
+         plr_data[plr.number] = {warehouses = {},
+                                 nr_heroes  = 0,
+                                 hero       = {}
+                                }
+         for j, w in pairs(plr.tribe.workers) do
+            if w.type_name == "soldier" then
+               plr_data[plr.number].hero = {w.max_health_level,
+                                            w.max_attack_level,
+                                            w.max_defense_level,
+                                            w.max_evade_level
+                                           }
+            end
+         end
+         local buildings = plr.tribe.buildings
+         local names = {}
+         for k, building in pairs(buildings) do
+            if building.type_name == "warehouse" then
+               table.insert(names, building.name)
+            end
+         end
+         plr_data[plr.number].warehouses = names
+         d_print("plr_data for: "..tostring(plr))
+         d_print("  warehouses:      "..t_print(plr_data[plr.number].warehouses))
+         d_print("  hero definition: "..t_print(plr_data[plr.number].hero))
+         d_print("  nr. of soldiers: "..tostring(plr_data[plr.number].nr_heroes))
+      end
+
+      -- set the objective with the game type for all players
+      broadcast_objective("win_condition", wc_descname, wc_desc)
+
+      -- Install statistics hook
+      hooks.custom_statistic = {
+         name = wc_first_hero,
+         pic = "images/wui/buildings/prefer_heroes.png",
+         calculator = function(p)
+            return plr_data[p.number].nr_heroes
+         end,
+      }
+
+      push_textdomain("first_heroes_win.wad", true)
```

Comment:
This needs to be *inside* the body of the functions `_get_statistic_msg` and `_get_result_msg` in which the translatable strings are located